### ISSUE TITLE
vaadin-spring-parent release 2.0.0 contains SNAPSHOT versions

### ISSUE THREAD

**Issue by psinghcredify** [vaadin/spring#217]
## vaadin-spring-parent
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <parent>
        <groupId>com.vaadin</groupId>
        <artifactId>vaadin-parent</artifactId>
        <version>1.0.0</version>
    </parent>

    <groupId>com.vaadin</groupId>
    <artifactId>vaadin-spring-parent</artifactId>
    <version>2.0.0</version>
    <packaging>pom</packaging>

<!-- These are typically overridden with BOMs -->
        <vaadin.version>8.0-SNAPSHOT</vaadin.version>

.....
```
## My pom.xml contains below:

```xml
<properties>
		<vaadin.version>8.0.0</vaadin.version>
	</properties>

	<dependencyManagement>
		<dependencies>
			<dependency>
				<groupId>com.vaadin</groupId>
				<artifactId>vaadin-bom</artifactId>
				<version>${vaadin.version}</version>
				<type>pom</type>
				<scope>import</scope>
			</dependency>
		</dependencies>
	</dependencyManagement>

<dependencies>

		<dependency>
			<groupId>com.vaadin</groupId>
			<artifactId>vaadin-spring-boot-starter</artifactId>
		</dependency>

...
```

With above, I still see maven trying to pull 8.0-SNAPSHOT version of vaadin-server.
It proves that below doesn't work:
```xml
<!-- These are typically overridden with BOMs -->
        <vaadin.version>8.0-SNAPSHOT</vaadin.version>
```

Is there any quick fix?

**Comment by mstahv** [comment]
Hi, for me the version is resolved correctly to the version declared in vaadin-bom. Could you share the full project or pom.xml?

**Comment by snicoll** [comment]
The bom is going to override those settings, that's why it's working for you. 

Try to build this

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>

	<groupId>com.example</groupId>
	<artifactId>demo</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<packaging>jar</packaging>

	<name>demo</name>
	<description>Demo project for Spring Boot</description>

	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
	</properties>

	<dependencies>
		<dependency>
			<groupId>com.vaadin</groupId>
			<artifactId>vaadin-spring-boot</artifactId>
			<version>2.0.1</version>
		</dependency>
	</dependencies>

</project>
```

You'll see that's downloading Vaadin 8.0 snapshot which is _super bad_. Also it looks like you have hard-coded the Sonatype OSS repo in the pom somewhere and published POMs should not have repositories.

**Comment by mstahv** [comment]
@hesara Could you fix the default version to some non-snapshot version? IIRC there was also some other fixes in queue, so maybe we could cut a release soon as well?

**Comment by snicoll** [comment]
You should also consider adding a validation goal in your release process to make sure such case is detected.

**Comment by psinghcredify** [comment]
for now this is what I have done to fix in my pom.xml

```xml
		<!-- Temporary Fix: vaadin-spring-parent has hardcoded the version to 8.0-SNAPSHOT -->
 		<dependency>
 			<groupId>com.vaadin</groupId>
 			<artifactId>vaadin-server</artifactId>
 			<version>8.0.5</version>
  		</dependency>
 
 		<dependency>
 			<groupId>com.vaadin</groupId>
 			<artifactId>vaadin-themes</artifactId>
 			<version>8.0.5</version>
 		</dependency>
 
 		<dependency>
 			<groupId>com.vaadin</groupId>
 			<artifactId>vaadin-client-compiled</artifactId>
 			<version>8.0.5</version>
 		</dependency>
 		<!-- Temporary Fix: End -->
```

**Comment by hesara** [comment]
This should be addressed by #219.
